roles/ood/tasks/main.yaml

```diff
@@ -78,6 +78,13 @@
       - { regexp: "^#?node_uri:.*$", replace: "node_uri: '/node'" }
       - { regexp: "^#?rnode_uri:.*$", replace: "rnode_uri: '/rnode'" }
 
+- name: update ood portal config for basic authentication
+  replace:
+    path: /etc/ood/config/ood_portal.yml
+    regexp: "^(#  - 'Require valid-user')"
+    replace: '\1\nauth:\n - "AuthType Basic"\n - "AuthName "private""\n - "AuthUserFile "/opt/rh/httpd24/root/etc/httpd/.htpasswd""\n' - "RequestHeader unset Authorization"\n' - "Require valid-user"\n'
+    backup: yes
+
```

Comment:
I have trouble executing this task. I'd suggest using `blockinfile` instead.

```
- name: update ood portal config for basic authentication
  blockinfile:
    path: /etc/ood/config/ood_portal.yml
    insertbefore: "#auth:"
    backup: yes
    block: |
      auth:
        - 'AuthType Basic'
        - 'AuthName "private"'
        - 'AuthUserFile "/opt/rh/httpd24/root/etc/httpd/.htpasswd"'
        - 'RequestHeader unset Authorization'
        - 'Require valid-user'
```

The result of above code:
```
# List of Apache authentication directives
# NB: Be sure the appropriate Apache module is installed for this
# Default: (see below, uses OIDC auth with Dex)
# BEGIN ANSIBLE MANAGED BLOCK
auth:
  - 'AuthType Basic'
  - 'AuthName "private"'
  - 'AuthUserFile "/opt/rh/httpd24/root/etc/httpd/.htpasswd"'
  - 'RequestHeader unset Authorization'
  - 'Require valid-user'
# END ANSIBLE MANAGED BLOCK
#auth:
#  - 'AuthType openid-connect'
#  - 'Require valid-user'
```